### PR TITLE
Replace comparisons with wrappers

### PR DESCRIPTION
**Pull request by radha-patel**
Replace `<`, `<=`, `>`, `>=`, and `=` involving index comparisons with appropriate masks.

### CODE REVIEW

**Comment by willow-ahrens** [comment]
is it possible to handle != as well, by putting a ! in front of the diagonal mask?

**Comment by willow-ahrens** [comment]
it looks like some of the tests for conversions with triangle levels have failed. could you take a look at the logs to see if you can reproduce?

**Comment by radha-patel** [comment]
> it looks like some of the tests for conversions with triangle levels have failed. could you take a look at the logs to see if you can reproduce?

Yep! I'm looking now

File: src/transforms/wrapperize.jl [hunk 1]
```diff
@@ -42,6 +42,40 @@ function get_wrapper_rules(alg, depth, ctx)
                 end
             end
         end),
+        (@rule call(<, ~i::isindex, ~j::isindex) => begin
+            if depth(i) < depth(j)
+                access(LoTriMask(), reader(), call(-, i, 1), j)
+            elseif depth(i) > depth(j)
+                access(UpTriMask(), reader(), call(+, i, 1), j)
+            end
+        end),
```
Comment (willow-ahrens) [inline src/transforms/wrapperize.jl:51]: Could we refactor each of these rules into two cases:
```
        (@rule call(<, ~i, ~j::isindex) => begin
            if depth(i) < depth(j)
                access(UpTriMask(), reader(), j, call(-, i, 1))
            end
        ),
        @rule call(<, ~i::isindex, ~j) => begin
            if depth(i) > depth(j)
                access(LoTriMask(), reader(), i, call(+, j, 1))
            end
        ),
 ```
 This code is different in a few ways:
 1. Only the inner (fast) index needs to be a vanilla index (as opposed to `i + 1`, etc.).
 2. This code flips the index order in order to always emit a column major access.

File: src/transforms/wrapperize.jl [hunk 2]
```diff
@@ -42,6 +42,40 @@ function get_wrapper_rules(alg, depth, ctx)
...
+            elseif depth(i) > depth(j)
+                access(LoTriMask(), reader(), i, j)
+            end
+        end),
+        (@rule call(==, ~i::isindex, ~j::isindex) => begin
+            access(DiagMask(), reader(), i, j)
+        end),
+        (@rule call(!=, ~i::isindex, ~j::isindex) => begin
+            call(!, access(DiagMask(), reader(), i, j))
+        end),
```
Comment (willow-ahrens) [inline src/transforms/wrapperize.jl:78]: These two rules are not always emitting column major code

**Review by willow-ahrens — CHANGES_REQUESTED** [review]
In addition to the more specific comments, none of these cases actually swap the order of i and j depending on which is bound first.

**Comment by radha-patel** [comment]
@willow-ahrens I've addressed your comments!

**Pull request merged**